### PR TITLE
Add forwarder for rollout status to frontend service

### PR DESCRIPTION
**Pull request by hannesg**
This commits adds a grpc-web forwarder to be used in the web frontend.

### CODE REVIEW

File: services/frontend-service/pkg/cmd/server.go [hunk 1]
```diff
@@ -166,26 +169,36 @@ func runServer(ctx context.Context) error {
...
-	con, err := grpc.Dial(c.CdServer, grpcClientOpts...)
+	cdCon, err := grpc.Dial(c.CdServer, grpcClientOpts...)
 	if err != nil {
 		logger.FromContext(ctx).Fatal("grpc.dial.error", zap.Error(err), zap.String("addr", c.CdServer))
 	}
+	var rolloutClient api.RolloutServiceClient = nil
+	if c.RolloutServer != "" {
+		rolloutCon, err := grpc.Dial(c.RolloutServer, grpcClientOpts...)
+		if err != nil {
+			logger.FromContext(ctx).Fatal("grpc.dial.error", zap.Error(err), zap.String("addr", c.RolloutServer))
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/pkg/cmd/server.go:180]:
```suggestion
			logger.FromContext(ctx).Fatal("grpc.dial.error could not connect to rollout service", zap.Error(err), zap.String("addr", c.RolloutServer))
```

File: services/frontend-service/pkg/cmd/server.go [hunk 2]
```diff
@@ -474,3 +488,22 @@ func (p *GrpcProxy) CreateEnvironment(
...
 	return p.EnvironmentClient.CreateEnvironment(ctx, in)
 }
+
+func (p *GrpcProxy) StreamStatus(in *api.StreamStatusRequest, stream api.RolloutService_StreamStatusServer) error {
+	if p.RolloutServiceClient == nil {
+		return status.Error(codes.Unimplemented, "rollout status not implemented")
+	}
+	if resp, err := p.RolloutServiceClient.StreamStatus(stream.Context(), in); err != nil {
+		return err
+	} else {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/pkg/cmd/server.go:498]: remove the `else` please

File: services/frontend-service/pkg/cmd/server.go [hunk 3]
```diff
@@ -474,3 +488,22 @@ func (p *GrpcProxy) CreateEnvironment(
...
+		for {
+			if item, err := resp.Recv(); err != nil {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/pkg/cmd/server.go:500]: should we add explicit tracing spans here, or is that too much?

File: services/frontend-service/pkg/cmd/server_test.go [hunk 4]
```diff
@@ -140,3 +144,71 @@ func TestServerHeader(t *testing.T) {
...
+						t.Logf("status: %q", res.StatusCode)
+						<-time.After(1 * time.Second)
+						continue
+					}
+					break
+				}
+				//
+				path, err := url.JoinPath("http://localhost:8081/", tc.RequestPath)
+				if err != nil {
+					panic(err)
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/pkg/cmd/server_test.go:189]: why not:
```suggestion
					t.Fatalf(err)
```
?

File: services/frontend-service/pkg/cmd/server.go (hunk 3 above)
Comment (hannesg) [reply services/frontend-service/pkg/cmd/server.go:500]: The underlying instrumentation already does this as for the overview service.

File: services/frontend-service/pkg/cmd/server.go (hunk 2 above)
Comment (hannesg) [reply services/frontend-service/pkg/cmd/server.go:498]: done

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**